### PR TITLE
6- /api/articles/:article_id/comments , update /api endpoint

### PR DESCRIPTION
**Pull request by faithcollison**

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -107,3 +107,41 @@ describe('GET /api/articles', () => {
...
+                    created_at: expect.any(String),
+                    author: expect.any(String),
+                    body: expect.any(String),
+                    article_id: expect.any(Number)
+                })
+            })
+            expect(response.body.articleComments).toBeSortedBy("created_at", {descending:true})
+        })
+    });
+    test('GET: 200 responds with an empty array if article_id exists but there are no snack with that id', () => {
```
Comment (aloush-dev) [inline __tests__/index.test.js]: snack?

File: controllers/articles.controller.js [hunk 2]
```diff
@@ -13,5 +13,20 @@ exports.getArticle = (req, res, next) => {
...
+}
+
+exports.getArticleComment = (req, res, next) => {
+    const {article_id} = req.params
+
+    const articlePromises = [selectArticleComment(article_id)]
+    if(article_id){
+        articlePromises.push(checkArticleExists(article_id))
+    }
+    Promise.all(articlePromises)
```
Comment (aloush-dev) [inline controllers/articles.controller.js]: If we add the checkArticleExists function to our promise.all then if the article doesn't exist that model will send a promise reject.
Would save having an if statement and pushing to an array.

File: models/articles.model.js [hunk 3]
```diff
@@ -18,4 +18,20 @@ exports.selectArticleById = (article_id) => {
...
+}
+
+exports.checkArticleExists = (article_id) => {
+    return db.query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id])
+    .then(({rows}) => {
+        if(rows.length === 0){
+            return Promise.reject({status:404, msg:"Article not found"})
+        }
+    })
+}
```
Comment (aloush-dev) [inline models/articles.model.js]: While this is completely fine, Is it not just an exact copy of the select ArticleById model?

**Review by aloush-dev — CHANGES_REQUESTED** [review]
Could you add a 400 bad request test also?

Check comments although the only thing that needs to change is the bad request test, Please resubmit once added.

Thanks

**Review by aloush-dev — CHANGES_REQUESTED** [review]
Could you add a 400 bad request test also?

Check comments although the only thing that needs to change is the bad request test, Please resubmit once added.

Thanks

**Review by MKD01 — APPROVED** [review]
Really good work, im happy for this to be merged

**Pull request merged**